app/controllers/item.js

```diff
@@ -138,18 +152,15 @@ export default Ember.Controller.extend({
     },
 
     setChildCategory(category) {
-      const parentId = category.get("parentId");
-      this.transitionToRoute(
-        "package_category",
-        parentId ? parentId : category.id
-      );
-      if (parentId) {
-        this.get("packageCategory").set("selectedCategoryId", category);
-      }
+      this.setAndRedirectToCategory(category);
     },
 
     back() {
-      window.history.back();
+      let prevPath = this.get("prevPath");
```

Comment:
would the opposite logic be possible ?

```
if (hasParentCategory) {
  return this.setAndRedirectToCategory(this.get("categoryObj"));
}
window.history.back();
```

allowing us to keep the `window.history.back()` , which we've been trying to implement.

Also if you plan on controlling the navigation yourself, you might want to enter the page using a `replace` instead of a `transition`. To avoid a large looping history

now the navigation history would look like `search_goods -> item -> search_goods`

Reply:
@patrixr I added `window.history.back()` some time back, but according to the requirement `window.history` doesn't suits here. Because when we click on `previous` and `next` buttons on item's page the `window.history` maintains its history and backlink malfunctions.
Case 1:
What I've done here is, if a child category is chosen, back action will take back to the parent category url with child category already chosen.
Also if on the previous

Reply:
And regarding the history, currently I have handled the history from both the pages.
If the user is on item page user will be redirected according to the logic I just explained in my above mentioned comment.

If user is on search page and he clicks on back or done, he'll be redirected to the browse-page
https://github.com/crossroads/browse.goodcity/pull/439/files#diff-44c533ec5932f27a50b79b2469b38784R77

Reply:
> `replace` instead of a `transition`

Actually i'm doing this transition in the controller, and replaceWith is not available in the controller, for that I'll have to access routing inside of a controller, but I don't think that's a good practice. So I avoided it @patrixr